linkerd/app/inbound/src/http/tests.rs

```diff
@@ -207,3 +306,41 @@ fn hello_server(
         Ok(io::BoxedIo::new(client_io))
     }
 }
+
+#[tracing::instrument]
+fn connect_error() -> impl Fn(Remote<ServerAddr>) -> io::Result<io::BoxedIo> {
+    move |_| {
+        Err(io::Error::new(
+            io::ErrorKind::Other,
+            "server is not listening",
+        ))
+    }
+}
+
+#[tracing::instrument]
+fn timeout_server(
+    http: hyper::server::conn::Http,
+) -> Box<dyn FnMut(Remote<ServerAddr>) -> ConnectFuture + Send> {
+    Box::new(move |endpoint| {
+        let http = http.clone();
+        let span = tracing::info_span!("timeout_server", ?endpoint);
+        Box::pin(
+            async move {
+                tracing::info!("sleeping so that the proxy hits a dispatch timeout");
+                tokio::time::sleep(std::time::Duration::from_secs(3)).await;
+                let (client_io, server_io) = support::io::duplex(4096);
+                let timeout_service =
+                    hyper::service::service_fn(|request: Request<Body>| async move {
+                        tracing::info!(?request);
+                        Ok::<_, io::Error>(Response::new(Body::from("timeout server")))
```

Comment:
TIOLI: it would probably be fine to make the `service_fn` just be `unreachable!()`, because we _should_ never send the server a request. up to you though.

Reply:
I replaced everything under the sleep with `unreachable!()`.